rr_computer_vision/src/endline_detection.cpp

```diff
@@ -0,0 +1,142 @@
+/** @file endline_detection.cpp
+ *  @brief Magenta endline detection
+ *
+ * Topics Subscribed:
+ *   /zed/rgb/image_rect_color
+ *
+ * Service called:
+ *   /Supervisor/count_lap
+ *
+ * @author Waleed Ahmed (w29ahmed)
+ * @author Yuchi(Allan) Zhao
+*/
+
+#include "endline_detection.hpp"
+#include "opencv2/highgui/highgui.hpp"
+#include "opencv2/imgproc/imgproc.hpp"
+#include <vector>
+
+// Constructor
+EndlineDetection::EndlineDetection(ros::NodeHandle nh) : it_(nh_)
+{
+  detection_status = false;
```

Comment:
declare these boys on how you declared the nodehandle; this will provide for a more consistent look on the code